pkg/hookbot/hookbot.go

```diff
@@ -310,16 +324,20 @@ func (h *Hookbot) Del(l Listener) {
 
 // The topic is everything after the "/pub/" or "/sub/"
 // Do not capture the "/unsafe". See note in `Topic()`.
-var TopicRE *regexp.Regexp = regexp.MustCompile("^(?:/unsafe)?/[^/]+/(.*)$")
+var TopicRE *regexp.Regexp = regexp.MustCompile("^(?:/unsafe)?/(?:pub|sub)/(.*)$")
```

Comment:
Sorry for the confusion, was just talking about the subset `/`, `/(pub|sub)/` and `/invalid/`. Obviously there is `/unsafe/` which makes this into six options:

- `/pub/foo`
- `/sub/foo`
- `/unsafe/pub/foo`
- `/unsafe/sub/foo`
- `/foo`

I think all of these cases should be handle individually and explicitly.